client/page.go

```diff
@@ -0,0 +1,126 @@
+package client
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+//Represents a page of records in a collection
+type Page struct {
+	Payload         map[string]interface{}
+	NextPageUrl     string
+	PreviousPageUrl string
+	Records         []interface{}
+	MetaKeys        map[string]bool
+	BaseUrl         string
+}
+
+func NewPage(baseUrl string, response *http.Response) *Page {
+	respBody := processResponse(response)
+	var page = Page{
+		Payload:  respBody,
+		MetaKeys: loadMetaKeys(),
+		BaseUrl:  baseUrl,
+	}
+
+	page.Records = page.loadPage()
+	page.NextPageUrl = page.getNextPageUrl()
+	page.PreviousPageUrl = page.getPreviousPageUrl()
+	return &page
+}
+
+func loadMetaKeys() map[string]bool {
+	return map[string]bool{
+		"end":               true,
+		"first_page_uri":    true,
+		"last_page_uri":     true,
+		"next_page_uri":     true,
+		"page":              true,
+		"page_size":         true,
+		"previous_page_uri": true,
+		"total":             true,
+		"num_pages":         true,
+		"start":             true,
+		"uri":               true,
+	}
+}
+
+func processResponse(response *http.Response) map[string]interface{} {
+	var data map[string]interface{}
+	if response != nil {
+		err := json.NewDecoder(response.Body).Decode(&data)
+
+		if err != nil {
+			return nil
+		}
+	}
+
+	return data
+}
+
+//Parses the collection of records out of a map payload and returns the records
+func (p *Page) loadPage() []interface{} {
+	if p.Payload["meta"] != nil && p.Payload["meta"].(map[string]interface{})["key"] != nil {
+		key := p.Payload["meta"].(map[string]interface{})["key"].(string)
+		return p.Payload[key].([]interface{})
+	} else {
+		var difference []string
+		for pk := range p.Payload {
+			if _, found := p.MetaKeys[pk]; found {
+				continue
+			}
+
+			difference = append(difference, pk)
+		}
+
+		if len(difference) == 1 {
+			return p.Payload[difference[0]].([]interface{})
+		}
```

Comment:
Needs a comment about what this is attempting to do. Note that in the CLI the strategy used is to look for a single property that's an array.

Reply:
oh, I like that idea.

Reply:
https://github.com/twilio/twilio-cli-core/blob/852477eac2d79b7cf9049d952771925711f5116c/src/services/twilio-api/twilio-client.js#L133-L143